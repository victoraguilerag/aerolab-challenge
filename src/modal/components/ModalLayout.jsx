import React from 'react'
import PropTypes from 'prop-types'
import coin from '../../icons/coin.svg'

export function ModalLayout (props) {
  const { handleModal, user: { name, points } } = props
  return (
    <div className="modal" style={styles.modal}>
      <div className="modal-container" style={styles.modalContainer}>
        <div className="user-data" style={styles.userData}>
          <p className="user-name" style={styles.userName}>{name}</p>
          <div className="user-points" style={styles.userPoints}>
            <p className="points" style={styles.points}>{points}</p>
            <img
              className="coin"
              style={styles.coin}
              width="24px"
              height="24px"
              src={coin}
              alt="coin"
            />
          </div>
          <div className="reedem-buttom" style={styles.reedemButton} onClick={() => handleModal('')}>
            <p className="reedem-text" style={styles.reedemText}>
              Return to catalog
            </p>
          </div>
        </div>
        {
          props.children
        }
      </div>
    </div>
  )
}

ModalLayout.propTypes = {
  handleModal: PropTypes.func.isRequired,
  user: PropTypes.shape({}).isRequired,
  children: PropTypes.element.isRequired
}

const styles = {
  'modal': {
    'userSelect': 'none',
    'position': 'fixed',
    'top': '0',
    'left': '0',
    'width': '100vw',
    'height': '100vh',
    'background': 'rgba(0,0,0,.5)',
    'display': 'flex',
    'justifyContent': 'center',
    'alignItems': 'center'
  },
  'modalContainer': {
    'width': '93%',
    'minHeight': '80%',
    'background': '#fff',
    'borderRadius': '25px',
    'display': 'flex',
    'alignItems': 'center',
    'justifyContent': 'center'
  },
  'userData': {
    'height': '80vh',
    'minWidth': '50%',
    'display': 'flex',
    'flexFlow': 'column',
    'justifyContent': 'center',
    'alignItems': 'center'
  },
  'userPoints': {
    'display': 'flex',
    'justifyContent': 'center',
    'alignItems': 'center',
    'background': '#ededed',
    'borderRadius': '100px',
    'width': '103px',
    'height': '45px'
  },
  'userName': {
    'fontSize': '24px',
    'color': '#616161',
    'letterSpacing': '-0.15px',
    'lineHeight': '48px',
    'textAlign': 'left'
  },
  'points': {
    'fontSize': '24px',
    'color': '#616161',
    'letterSpacing': '-0.15px',
    'lineHeight': '48px',
    'textAlign': 'left'
  },
  'reedemButton': {
    'backgroundImage': 'linear-gradient(-180deg, #0ad4fa 0%, #25bbf1 100%)',
    'width': '228px',
    'height': '42px',
    'margin': '40px auto',
    'borderRadius': '5px',
    'display': 'flex',
    'justifyContent': 'center',
    'alignItems': 'center',
    'cursor': 'pointer'
  },
  'reedemText': {
    'fontSize': '18px',
    'color': 'white',
    'letterSpacing': '-0.04px',
    'textAlign': 'center'
  }
}

export default ModalLayout
