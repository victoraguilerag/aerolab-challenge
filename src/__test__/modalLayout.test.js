import React from 'react'
import { ModalLayout } from '../modal/components/ModalLayout.jsx'
import renderer from 'react-test-renderer'
import userStub from '../stub/user.js'

const functionMock = () => console.log('function mock');

test('Modal Layout renders', () => {
  const component = renderer.create(
    <ModalLayout
      handleModal={functionMock}
      user={userStub}
    >
      <div />
    </ModalLayout>,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
