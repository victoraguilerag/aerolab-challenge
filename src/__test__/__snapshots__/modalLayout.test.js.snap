// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Layout renders 1`] = `
<div
  className="modal"
  style={
    Object {
      "alignItems": "center",
      "background": "rgba(0,0,0,.5)",
      "display": "flex",
      "height": "100vh",
      "justifyContent": "center",
      "left": "0",
      "position": "fixed",
      "top": "0",
      "userSelect": "none",
      "width": "100vw",
    }
  }
>
  <div
    className="modal-container"
    style={
      Object {
        "alignItems": "center",
        "background": "#fff",
        "borderRadius": "25px",
        "display": "flex",
        "justifyContent": "center",
        "minHeight": "80%",
        "width": "93%",
      }
    }
  >
    <div
      className="user-data"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexFlow": "column",
          "height": "80vh",
          "justifyContent": "center",
          "minWidth": "50%",
        }
      }
    >
      <p
        className="user-name"
        style={
          Object {
            "color": "#616161",
            "fontSize": "24px",
            "letterSpacing": "-0.15px",
            "lineHeight": "48px",
            "textAlign": "left",
          }
        }
      >
        Daniela Veliz
      </p>
      <div
        className="user-points"
        style={
          Object {
            "alignItems": "center",
            "background": "#ededed",
            "borderRadius": "100px",
            "display": "flex",
            "height": "45px",
            "justifyContent": "center",
            "width": "103px",
          }
        }
      >
        <p
          className="points"
          style={
            Object {
              "color": "#616161",
              "fontSize": "24px",
              "letterSpacing": "-0.15px",
              "lineHeight": "48px",
              "textAlign": "left",
            }
          }
        >
          2000
        </p>
        <img
          alt="coin"
          className="coin"
          height="24px"
          src={Object {}}
          width="24px"
        />
      </div>
      <div
        className="reedem-buttom"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundImage": "linear-gradient(-180deg, #0ad4fa 0%, #25bbf1 100%)",
            "borderRadius": "5px",
            "cursor": "pointer",
            "display": "flex",
            "height": "42px",
            "justifyContent": "center",
            "margin": "40px auto",
            "width": "228px",
          }
        }
      >
        <p
          className="reedem-text"
          style={
            Object {
              "color": "white",
              "fontSize": "18px",
              "letterSpacing": "-0.04px",
              "textAlign": "center",
            }
          }
        >
          Return to catalog
        </p>
      </div>
    </div>
    <div />
  </div>
</div>
`;
